lib/puppet/provider/iis_mimetype/iis_mimetype.rb

```diff
@@ -0,0 +1,156 @@
+require File.join(File.dirname(__FILE__), '../../../puppet/provider/iis_powershell')
+
+Puppet::Type.type(:iis_mimetype).provide(:iis_mimetype, parent: Puppet::Provider::IIS_PowerShell) do
+  confine    :iis_version     => ['7.0', '7.5', '8.0', '8.5', '10.0']
+  confine    :operatingsystem => :windows
+  defaultfor :operatingsystem => :windows
+
+  commands :powershell => PuppetX::IIS::PowerShellCommon.powershell_path
+
+  desc <<-EOT
+    Manage IIS unique MIME types in the collection of static content types.
+
+    Example:
+    iis_mimetype {'.zip':
+      mimetype => 'application/x-zip-compressed',
+    }
+  EOT
+
+  mk_resource_methods
+
+  def initialize(value={})
+    super(value)
+    @property_flush = {}
+  end
+
+  def self.get_mimetypes
+    cmd = "Get-WebConfiguration -Recurse -Filter /system.webServer/staticContent/mimeMap | select fileExtension,mimeType,Location,PSPath | ConvertTo-Json"
+    result = run(cmd)
+
+    #cmd = ps_script_content('_getvirtualdirectories', @resource)
+    #result   = run(cmd)
+    return [] if result.nil?
+
+    json = self.parse_json_result(result[:stdout])
```

Comment:
@jpogran When you say "trying to parse the return which was already parsed into a ruby hash" I assume you mean this line. But if you look [here](https://github.com/puppetlabs/puppetlabs-iis/blob/master/lib/puppet/provider/iis_site/webadministration.rb#L119) and [here](https://github.com/puppetlabs/puppetlabs-iis/blob/master/lib/puppet/provider/iis_virtual_directory/webadministration.rb#L100), you see the existing providers are doing exactly the same.